### PR TITLE
Bunch of fixes/improvements for hardware wallet connector API

### PR DESCRIPTION
**Pull request by yushih**
1. fix Trezor signing problems reported by Denis
2. show "unsupported" message when the dApp connects with authentication and the user picks a hardware wallet
3. minor fixes and improvements

### CODE REVIEW

**Comment by neuodev** [comment]
[autoscript] ⚠️ `yushi/hw-wallet-connector-2` doesn't full our branch naming convention.
Should follow `<tag>/<jira-ticket-id>/<short-name>`
<details>
<summary>Details</summary>

- `<tag>`: `fix`, `feat`, `chore`, `refactor`, `test`, ...
- `<jira-ticket-id>`: The issue ID of JIRA all capital letters
- `<short-name>`: A short and descriptive name for the branch
</details>

[](__JIRA_COMMENT_FLAG)

File: packages/yoroi-connector/example-cardano/index.js [hunk 1]
```diff
@@ -480,6 +480,18 @@ signTx.addEventListener("click", () => {
...
+      const amount2 = BigInt(utxo2.amount);
+      if (amount1 > amount2) {
+        return -1;
+      }
+      if (amount1 < amount2) {
+        return 1;
+      }
+      return 0;
+    });
     const utxo = utxos[0];
```
Comment (neuodev) [inline packages/yoroi-connector/example-cardano/index.js]:
```suggestion
    // pick the UTXO with the most ADAs
    const utxo = utxos.reduce((prev, curr) => BigInt(prev.amount) > BigInt(curr.amount) ? prev : curr);
```

**Review by a-hosani — APPROVED** [review]

File: packages/yoroi-connector/example-cardano/index.js (hunk 1 above)
Comment (yushih) [reply packages/yoroi-connector/example-cardano/index.js]: Changed as suggested. Thanks.

Comment (neuodev) [reply packages/yoroi-connector/example-cardano/index.js]: Thanks @yushih !

**Review by Nebyt — APPROVED** [review]

**Pull request merged**